sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,326 @@
-  // Number of receipts
-  int numReceipts;
-  // Indexs of empty slots caused by the function `withdraw`
-  mapping (int => int) emptySlotsStack;
+  mapping (int => bytes32) shardead;
+
+  // Number of notaries
+  uint public notary_pool_len;
+  // Indexes of empty slots caused by the function `withdraw`
+  uint[] empty_slots_stack;
   // The top index of the stack in empty_slots_stack
-  int emptySlotsStackTop;
-  // Has the collator deposited before?
-  mapping (address => bool) public isCollatorDeposited;
+  uint empty_slots_stack_top;
+
+  // Notary sample size at current period and next period
+  uint current_period_notary_sample_size;
+  uint next_period_notary_sample_size;
+  uint sample_size_last_updated_period;
 
   // Constant values
-  uint constant periodLength = 5;
-  int constant public shardCount = 100;
-  // The exact deposit size which you have to deposit to become a collator
-  uint constant depositSize = 1000 ether;
+  uint constant PERIOD_LENGTH = 5;
+  // Exact collation body size (1mb)
+  uint constant COLLATION_SIZE = 2 ** 20;
+  // Number of shards
+  uint constant SHARD_COUNT = 100;
+  // The minimum deposit size for a notary
+  uint constant NOTARY_DEPOSIT = 1000 ether;
+  // The reward for notary on voting for a collation
+  uint constant NOTARY_REWARD = 0.001 ether;
+  // The minimum deposit size for a proposer
+  uint constant PROPOSER_DEPOSIT = 1 ether;
+  // The minimum balance of a proposer (for notaries)
+  uint constant MIN_PROPOSER_BALANCE = 0.1 ether;
+  // Time the ether is locked by notaries (Not constant for testing)
+  uint NOTARY_LOCKUP_LENGTH = 16128;
+  // Time the ether is locked by proposers (Not constant for testing)
+  uint PROPOSER_LOCKUP_LENGTH = 48;
   // Number of periods ahead of current period, which the contract
-  // is able to return the collator of that period
-  uint constant lookAheadPeriods = 4;
+  // is able to return the notary of that period
+  uint constant LOOKAHEAD_LENGTH = 4;
+  // Number of notaries to select from notary pool for each shard in each period
+  uint constant COMMITTEE_SIZE = 135;
+  // Threshold(number of notaries in committee) for a proposal to be deem accepted
+  uint constant QUORUM_SIZE = 90;
 
   // Log the latest period number of the shard
-  mapping (int => int) public periodHead;
+  mapping (int => int) public period_head;
 
-  function SMC() public {
+  function SMC(uint notary_lockup_length, uint proposer_lockup_length) public {
+    NOTARY_LOCKUP_LENGTH = notary_lockup_length;
+    PROPOSER_LOCKUP_LENGTH = proposer_lockup_length; 
   }
 
-  // Returns the gas limit that collations can currently have (by default make
-  // this function always answer 10 million).
-  function getCollationGasLimit() public pure returns(uint) {
-    return 10000000;
-  }
-
-  // Uses a block hash as a seed to pseudorandomly select a signer from the collator pool.
-  // [TODO] Chance of being selected should be proportional to the collator's deposit.
+  event LOG(uint L);
```

Comment:
This event is just called LOG?

Reply:
It has no any usage, was using it to debug, you can remove it